lib/chains/saveScreenshotChain.js

```diff
@@ -0,0 +1,58 @@
+const makeChain = require('../utils/makeChain');
+const {
+	makeToStringComposer,
+	makeThenComposer,
+	makeToJSONComposer,
+	abandonComposer,
+} = require('../composers');
+const t = require('../texts');
+const {getOpType} = require('../utils/opType');
+const {processServerResponse} = require('../utils/socketChainHelper');
+const logger = require('../utils/logger');
+const {validate, validators} = require('../validation');
+
+const toJSON = () => ({type: 'takeScreenshot'});
+const toString = t.saveScreenshot;
+const beforeSendMsg = (data) => logger.log(
+	getOpType(data),
+	toString(),
+);
+
+const toStringComposer = makeToStringComposer(toString, toJSON);
+const thenComposer = makeThenComposer(toJSON, processServerResponse(toString), beforeSendMsg);
+const toJSONComposer = makeToJSONComposer(toJSON);
+
+const getComposers = (data) => {
+	const output = [
+		toStringComposer,
+		thenComposer,
+		toJSONComposer,
+	];
+
+	if (!data.isAbandoned) {
+		output.push(abandonComposer);
+	}
+
+	return output;
+};
+
+/**
+ * @param {string} fileName
+ */
+const saveScreenshotChain = (fileName) => makeChain(getComposers, {
+	type: 'takeScreenshot',
+	fileName: validate(
+		validators.NON_EMPTY_STRING,
+		fileName,
```

Comment:
@JeKachan we discussed file extension, I think we agreed that if file name ends with `.png` - we use it without changes. If it ends with anything else - we append `.png` to the filename.

`my-image.png` => `my-image.png`
`my-image.jpeg` => `my-image.jpeg.png`

By the way, are you sure it's always in PNG format?

Reply:
Discussed this in Slack